google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/resources/RepositorySelector.java

```diff
@@ -0,0 +1,256 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.resources;
+
+import com.google.cloud.tools.intellij.login.CredentialedUser;
+import com.google.cloud.tools.intellij.ui.CustomizableComboBox;
+import com.google.cloud.tools.intellij.ui.CustomizableComboBoxPopup;
+import com.google.cloud.tools.intellij.ui.GoogleCloudToolsIcons;
+
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.application.ModalityState;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.ui.popup.ComponentPopupBuilder;
+import com.intellij.openapi.ui.popup.JBPopup;
+import com.intellij.openapi.ui.popup.JBPopupFactory;
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.ui.awt.RelativePoint;
+import com.intellij.ui.components.JBScrollPane;
+
+import org.jetbrains.annotations.Nullable;
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Desktop;
+import java.awt.Dimension;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Enumeration;
+
+import javax.swing.BorderFactory;
+import javax.swing.Box;
+import javax.swing.BoxLayout;
+import javax.swing.JButton;
+import javax.swing.JPanel;
+import javax.swing.JTree;
+import javax.swing.tree.DefaultMutableTreeNode;
+import javax.swing.tree.DefaultTreeModel;
+import javax.swing.tree.TreeNode;
+import javax.swing.tree.TreeSelectionModel;
+
+/**
+ * Defines a repository selector UI widget allowing the user to select a repository associated
+ * with a given Google Cloud project.
+ */
+public class RepositorySelector extends CustomizableComboBox implements CustomizableComboBoxPopup {
+
+  private static final Logger logger = Logger.getInstance(RepositorySelector.class);
+
+  private JBPopup popup;
+  private RepositoryPanel panel;
+  private String cloudProject;
+  private CredentialedUser user;
+  private boolean canCreateRepository;
+  private static final int SELECTOR_HEIGHT = 140;
+  private static final int SELECTOR_WIDTH = 400;
+
+
+  public RepositorySelector(@Nullable String cloudProject,
+      @Nullable CredentialedUser user, boolean canCreateRepository) {
+    this.cloudProject = cloudProject;
+    this.user = user;
+    this.canCreateRepository = canCreateRepository;
+
+    getTextField().getEmptyText().setText("Enter a cloud repository ID...");
+  }
+
+  @Nullable
+  public String getSelectedRepository() {
+    if (StringUtil.isEmpty(getText())) {
+      return null;
+    }
+
+    Enumeration repos = panel.getRepositories().children();
+    while (repos.hasMoreElements()) {
```

Comment:
We could use Collections.list() to transform the enumeration into a collection. Would the extra pass be worth simpler code? I'm leaning to think yes, but I don't feel strongly about this.